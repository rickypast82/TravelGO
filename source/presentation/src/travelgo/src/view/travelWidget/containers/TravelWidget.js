//@flow
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { type State} from '../../../lib/redux-helper';
import { 
    type DestinationsT,
    getAllDestinations 
} from '../../../modules/travel';
import TravelCard from '../components/TravelCard';

//Prova da cancellare
import { DESTINATIONS } from './test/mockeddata';

type StateProps = {|
    destinations:DestinationsT
|};

const ContainerDiv = styled.div``;

const TravelWidgetContainer = ({destinations}: StateProps) => { 

    return (
        <ContainerDiv id='TravelWidgetContainerDiv'>
            {destinations.map(item => (
                <TravelCard key={item.id} 
                    id={item.id}
                    destination={item.destination} 
                    departureDate={item.departureDate} 
                    note={item.note}
                    returnDate={item.returnDate}
                    imageSrc='Roma.jpg'/>))}
        </ContainerDiv>
    )
};
const mapStateToProps = (state:State):StateProps => ({
    destinations: DESTINATIONS//getAllDestinations(state)
});


export { TravelWidgetContainer, mapStateToProps };
export default connect(mapStateToProps)(TravelWidgetContainer);