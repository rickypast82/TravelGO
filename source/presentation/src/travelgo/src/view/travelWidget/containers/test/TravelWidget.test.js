//@flow
/*eslint no-magic-numbers: "off"*/
import React from 'react';
import { shallow } from 'enzyme';
import { TravelWidgetContainer } from '../TravelWidget';
import { DESTINATIONS } from './mockeddata';
describe('TravelWidgetContainer  test', () => {
 
    it('renders without crashing with all props', () => {
        const wrapper = shallow( 
            <TravelWidgetContainer destinations={DESTINATIONS} />);
        expect(wrapper).toMatchSnapshot();
    });
/*
    it('renders without crashing with required props', () => {
        const wrapper = shallow( 
            <TravelCard 
                id={1}
                destination='Roma' 
                note='note roma' 
                imageSrc='Roma.jpg'
            />);
        expect(wrapper).toMatchSnapshot();
    });
    */

});
