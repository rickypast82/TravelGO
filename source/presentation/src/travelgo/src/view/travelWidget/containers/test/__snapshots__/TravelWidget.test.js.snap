// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelWidgetContainer  test renders without crashing with all props 1`] = `
<styled.div
  id="TravelWidgetContainerDiv"
>
  1
  <TravelCard
    departureDate={100}
    destination="Rome"
    id={1}
    imageSrc="Roma.jpg"
    key="1"
    note="note di roma"
    returnDate={200}
  />
  2
  <TravelCard
    departureDate={300}
    destination="Venice"
    id={2}
    imageSrc="Roma.jpg"
    key="2"
    note="note di venezia"
    returnDate={400}
  />
</styled.div>
`;
